packages/core-ethereum/src/ethereum.ts

```diff
@@ -66,6 +66,28 @@ export async function createChainWrapper(providerURI: string, privateKey: Uint8A
     durations.minutes(15)
   )
 
+  // naive implementation, assumes transaction is not replaced
+  // temporary used until https://github.com/ethers-io/ethers.js/issues/1479
+  // is fixed
+  async function waitForConfirmations(
+    transactionHash: string,
+    confimations: number,
+    timeout: number
+  ): Promise<providers.TransactionResponse> {
+    let started = 0
+    let response: providers.TransactionResponse
+
+    while (started < timeout) {
+      response = await provider.getTransaction(transactionHash)
+      if (response && response.confirmations >= confimations) break
+      // wait 1 sec
+      await new Promise((resolve) => setTimeout(resolve, 1e3))
```

Comment:
why do we need the timeout?

Reply:
https://docs.ethers.io/v5/api/providers/provider/#Provider-getTransaction it will return `null` if it doesn't find it